File: Programmers/Lv4/Lv4_쿠키구입.py
def solution(cookie):
    answer = 0
    sum = []
    sum.append(0)
    sum.append(cookie[0])
    for i in range(1, len(cookie)):
        sum.append(sum[i] + cookie[i])

    size = len(sum)
    start_sum = 0

    for start in range(size):
        for end in range(size - 1, start, -1):
            total = sum[end] - sum[start]
            if total % 2 != 0:  # 荑좏궎媛 ??섍컻?대㈃ ?⑥뒪
                continue
            if total // 2 < answer:  # ?섎닠 媛?몃룄 ?댁쟾 ?듬낫???묒쓣 寃쎌슦 ?⑥뒪
                continue
            for index in range(end, start, -1):
                if sum[index] - sum[start] == total // 2:
                    if total // 2 > answer:
                        answer = total // 2
                    break
                elif sum[index] - sum[start] < total // 2:
                    break
    return answer

File: Programmers/Lv4/Lv4_쿠키구입.cpp
#include <string>
#include <vector>
using namespace std;

int solution(vector<int> cookie) {
	int answer = 0;
	int size = cookie.size();
	vector<int> sum;		// 1 2 4 7 
	sum.push_back(0);
	sum.push_back(cookie[0]);
	for (int i = 1; i < size; i++)
		sum.push_back(sum[i] + cookie[i]);
	
	size = sum.size();

	int start_sum = 0;
	for (int start = 0; start < size; start++) {
		for (int end = size-1; end > start; end--) {
			int total = sum[end] - sum[start];
			if (total % 2 != 0)				// 쿠키가 홀수개이면 패스
				continue;
			if (total / 2 < answer)			// 나눠 가져도 이전 답보다 작을 경우 패스
				continue;
			for (int index = end; index > start; index-- ) {
				if (sum[index] - sum[start] == total / 2) {
					if (total / 2 > answer)
						answer = total / 2;
					break;
				}
				else if (sum[index] - sum[start] < total / 2)
					break;
			}
		}
	}

	return answer;
}
